contrib/cirrus/setup.sh

```diff
@@ -18,3 +18,22 @@ msg "************************************************************"
 msg "Setting up runtime environment"
 msg "************************************************************"
 show_env_vars
+
+if [[ "$1" == "cross" ]]; then
+    # This is required by rustup-init (it will complain if any tools are in $PATH)
+    rust_version=$(rpm -q --qf '%{v}' rust)
+    msg "Replacing Fedora native rust toolchain with upstream '$rust_version'"
+    dnf erase -y cargo rust rust-std-static
+
+    msg "Initializing upstream rust environment."
+    mkdir -p $CARGO_HOME/bin/
+    cd $CARGO_HOME/bin/
+    curl --fail --silent --location -O --url \
+        "https://static.rust-lang.org/rustup/dist/x86_64-unknown-linux-gnu/rustup-init"
+    chmod +x rustup-init
+    # complete_setup() will take care of updating $PATH properly
+    showrun ./rustup-init -y \
+        --default-toolchain "$rust_version" \
+        --no-modify-path \
+        --target $(tr ' ' ','<<<"$CROSS_TARGETS")
+fi
```

Comment:
@lsm5 and @Luap99 PTAL, does this seem really fragile to you?  (it does to me)

WDYT if instead of using the Fedora RPMs at all, I installed upstream rust into the VMs at image build time?

I think as long as we pick some static rust toolchain version equal-or-less than the current Fedora version, it's reasonably safe?  And we gain access to `rustup` (unavailable in Fedora).

A down-side could be...it will probably make building netavark/aardvark RPMs and SRPMs harder (since we're using non-RPM toolchain and dependencies).  Possibly making development harder (need to install custom toolchain).  OTOH, this is ***upstream*** testing, so perhaps staying in sync with the distro. is less important?

Opinions?